### ISSUE TITLE
Syntax error when using Unicode escape sequence with curly brackets in a string

### ISSUE THREAD

**Issue by jevancc** [boa-dev/boa#827]
**Describe the bug**
The behavior of Unicode code point escape (Unicode escape with curly brackets) is different from the spec.

**To Reproduce**
```
"\u{AA}"       // syntax error, should pass
"\u{{AA}}"     // pass, should throw syntax error
"\u{{AA}"      // freeze, should throw syntax error
"\u{AA}}"      // freeze, should pass
```

**Expected behavior**
The string should be tokenized and handled correctly. The expected behavior can be found in [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#Escape_notation).
Some lexer tests should also be updated.

**Build environment (please complete the following information):**

- OS: `Windows 10`
- Version: `OS Build 190941.508`
- Target triple: `x86_64-pc-windows-msvc`
- Rustc version: `rustc 1.49.0-nightly (91a79fb29 2020-10-07)`

**Additional context**
It should be bugs in [`string.rs`](https://github.com/boa-dev/boa/blob/master/boa/src/syntax/lexer/string.rs#L125) and [`cursor.rs`](https://github.com/boa-dev/boa/blob/master/boa/src/syntax/lexer/cursor.rs).

**Comment by Razican** [comment]
Hi! Is this solved by #826?

**Comment by jevancc** [comment]
> Hi! Is this solved by #826?

No, it's not. Seems to be an independent bug in `string.rs` so I created this issue for it :)
